ocaml/networkd/bin/network_server.ml

```diff
@@ -1474,10 +1474,21 @@ end
 module PVS_proxy = struct
   open S.PVS_proxy
 
-  let path = ref "/run/pvsproxy"
+  let path = ref ""
+
+  let depriv_path = "/run/pvsproxy-state/socket"
+
+  let legacy_path = "/opt/citrix/pvsproxy/socket/pvsproxy"
+
+  let default_path () =
+    if Sys.file_exists depriv_path then
+      depriv_path
+    else
+      legacy_path
 
   let do_call call =
-    try Jsonrpc_client.with_rpc ~path:!path ~call ()
+    let p = match !path with "" -> default_path () | path -> path in
+    try Jsonrpc_client.with_rpc ~path:p ~call ()
```

Comment:
Does the path get created by the rpms? I would prefer if the default for `path` wasn't empty and could be calculated when the module gets loaded, at networkd start:

```suggestion
  let depriv_path = "/run/pvsproxy-state/socket"

  let legacy_path = "/opt/citrix/pvsproxy/socket/pvsproxy"

  let default_path () =
    if Sys.file_exists depriv_path then
      depriv_path
    else
      legacy_path

  let path = ref (default_path ())

  let do_call call =
    try Jsonrpc_client.with_rpc ~path:!path ~call ()
```

Reply:
I don't like anything non-trivial getting done at module load time, as it opens the door for race conditions. In this case, there is the code that updates `path` from the config file.

Also, in another thread we concluded that we need to check each time, for now.

Reply:
Why can't we call `legacy_path()` in `do_call` to avoid the problem of file operations at module load time?